core/src/vote_redundancy_checker.rs

```diff
@@ -0,0 +1,146 @@
+use log::*;
+use solana_runtime::bank::Bank;
+use solana_sdk::{clock::Slot, pubkey::Pubkey, transaction::Transaction};
+use solana_vote_program::vote_transaction;
+use std::{collections::HashMap, sync::Arc};
+
+#[derive(Default, Debug)]
+pub struct VoteRedundancyChecker {
+    latest_voted_slots: HashMap<Pubkey, Slot>,
```

Comment:
If you make this a `DashMap<Pubkey, AtomicU64>`, you can get rid of the lock around this `VoteRedundancyChecker`, which should be a lot more performant.